src/main.c

```diff
@@ -208,6 +211,21 @@ static uint16_t common_tasks(tmosTaskID task_id, uint16_t events)
 		return events ^ BLE_NEXT_STEP;
 	}
 
+	if (events & DETECT_CHARGING) {
+		static int was_charging = 0;
+		int is_charging = charging_status();
+
+		if (is_charging && !was_charging) {
+			mode = BOOT;
+			disp_charging();

```

Comment:
This fixes the leftover lit pixels:
```suggestion
                        fb_fill(fb, 0);
			disp_charging();
```